File: apps/ios/Sources/QuranRecitationApp/DesignSystem/Components/BrandCard.swift
import SwiftUI

/// Visual style for `BrandCard`. Top-level enum so callers can pass
/// the value through helper functions without locking the generic
/// `Content` type to a specific witness.
enum BrandCardStyle {
  case standard
  case inverse
}

/// Rounded card surface used for ayah cards, library rows, and result
/// panels. Default style is cream-on-cream with a subtle shadow;
/// `.inverse` flips to the dark variant used for the Continue card and
/// Recording panel.
struct BrandCard<Content: View>: View {
  private let style: BrandCardStyle
  private let content: Content

  init(style: BrandCardStyle = .standard, @ViewBuilder content: () -> Content) {
    self.style = style
    self.content = content()
  }

  var body: some View {
    content
      .padding(Spacing.lg)
      .background(background)
      .clipShape(RoundedRectangle(cornerRadius: Spacing.cardCornerRadius, style: .continuous))
      .shadow(color: shadowColor, radius: 8, x: 0, y: 2)
  }

  private var background: Color {
    switch style {
    case .standard: return Color.brand.card
    case .inverse:  return Color.brand.cardInverse
    }
  }

  private var shadowColor: Color {
    switch style {
    case .standard: return Color.black.opacity(0.04)
    case .inverse:  return Color.black.opacity(0.12)
    }
  }
}

File: apps/android/src/main/java/com/tilawah/android/designsystem/components/BrandCard.kt
package com.tilawah.android.designsystem.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.tilawah.android.designsystem.BrandTheme

/**
 * Rounded card surface used for ayah cards, library rows, and result
 * panels. Default style is cream-on-cream with a subtle shadow;
 * [BrandCardStyle.Inverse] flips to the dark variant used for the
 * Continue card and Recording panel.
 *
 * Mirrors `apps/ios/Sources/QuranRecitationApp/DesignSystem/Components/BrandCard.swift`.
 */
@Composable
fun BrandCard(
    modifier: Modifier = Modifier,
    style: BrandCardStyle = BrandCardStyle.Standard,
    border: BorderStroke? = null,
    content: @Composable () -> Unit,
) {
    val colors = BrandTheme.colors
    val spacing = BrandTheme.spacing
    val background = when (style) {
        BrandCardStyle.Standard -> colors.card
        BrandCardStyle.Paper -> colors.paper
        BrandCardStyle.Inverse -> colors.cardInverse
    }
    val shadowElevation = when (style) {
        BrandCardStyle.Standard -> 2.dp
        BrandCardStyle.Paper -> 2.dp
        BrandCardStyle.Inverse -> 6.dp
    }
    val shape = RoundedCornerShape(spacing.cardCornerRadius)
    Surface(
        modifier = modifier
            .clip(shape)
            .background(background)
            .then(if (border != null) Modifier.border(border, shape) else Modifier),
        color = Color.Transparent,
        shadowElevation = shadowElevation,
    ) {
        androidx.compose.foundation.layout.Box(
            modifier = Modifier.padding(spacing.lg),
        ) {
            content()
        }
    }
}
